### DISCUSSION TITLE
Including a code-chunk title when `code-summary` is defined.

### DISCUSSION THREAD

**Discussion by JoFrhwld** [quarto-dev/quarto-cli#6799] [Feature Requests]
### Description

I've started using `filename` to add brief descriptive titles to my code chunks, which I think is helpful for code legibility. However, it seems like a misuse given the name of the option.

It would be nice if, instead, when the code chunk options are

```
#| code-fold: false
#| code-summary: "description"
```

The value given to `code-summary` was styled like the current `filename` option in the output. At the moment, the value passed to `code-summary` only appears in the output when `code-fold` is `true`.

**Comment by JoFrhwld** [comment]
This would also clean up how code chunk titles and code folding interact:

### Currently:

```
#| filename: "description"
#| code-fold: true
```

- Code folding is ignored
- Code chunk appears with a title

```
#| filename: "description"
#| code-fold: false
```

- Code chunk appears with a title

### With proposed feature

```
#| code-summary: "description"
#| code-fold: true
```

- Code-folding and summary are style as they are currently.

```
#| code-summary: "description"
#| code-fold: false
```

- Code chunk appears with a title

**Reply by mcanouil** [reply]
Filename is not the same thing as code-summary and I don't think they should.

If your concerns are about the style used for the details HTML tag around code blocks, you can customise that yourself.

See for example: https://github.com/EmilHvitfeldt/quarto-revealjs-codewindow

**Reply by JoFrhwld** [reply]
Agreed that `filename` shouldn't be the same thing as `code-summary`. At the moment I'm *misusing* `filename` to give descriptive titles to code chunks, as in this screenshot:

![titled_code_chunk](https://github.com/quarto-dev/quarto-cli/assets/178755/bf6c0621-c3b7-4ec7-a125-18f62acc2768)

I actually got the idea from [the quarto docs](https://quarto.org/docs/output-formats/pdf-engine.html#installing-tex) ([source](https://github.com/quarto-dev/quarto-web/blob/ca30d7cbbfbd70bb9172a63ce1c81c011ac86693/docs/output-formats/pdf-engine.qmd#L18-L20)). I would, ideally, put descriptive titles on code chunks using `code-summary` instead.

**Reply by mcanouil** [reply]
`code-summary` refers to the summary HTML tag of a details tag which is what allows code folding.

Note that there is the code listing feature: https://quarto.org/docs/authoring/cross-references.html#code-listings

And extension is also a good way to get whatever you want in whatever shape you want as the example I already shared which could be mixed with code folding.

**Reply by JoFrhwld** [reply]
It doesn't seem like code listing & list captions can be set within an executed code-chunk.

Realistically, what I'll do is just keep adding descriptive titles to code chunks using `filename`. I actually think it looks fine, so I don't need to dig into the styling. I just thought the semantics weren't quite right for how I was using it.
